### ISSUE TITLE
add `ElementOf<T extends any[]>`

### ISSUE THREAD

**Issue by quezak** [krzkaczor/ts-essentials#148]
We already have `ValueOf<T>` for objects, I thought it would be nice to have a similar helper for array element type, since `ValueOf` with arrays includes more than just the elements [1].

Why I'm doing this via an issue and not a PR is because I have two seemingly equal implementations, does anyone know any advantages of either of those? (like performance, better results for edge cases, better error messages, etc)

```typescript
type ElementOf<T extends any[]> = T extends (infer ET)[] ? ET : never;
type ElementOf<T extends any[]> = T[number];
```

[1] 
```typescript
const1 a1 = [true, false];
type A1ET = ValueOf<typeof arr1>`;

// results:
type A1ET = number | boolean | (() => string) | (() => string) | (() => boolean | undefined) | ((...items: boolean[]) => number) | {
    (...items: ConcatArray<boolean>[]): boolean[];
    (...items: (boolean | ConcatArray<boolean>)[]): boolean[];
} | ... 23 more ... | ((searchElement: boolean, fromIndex?: number | undefined) => boolean)
```

**Comment by krzkaczor** [comment]
I think simply (1) is post conditional types and (2) is pre ;) I mean it was possible to do so before but only `infer` made it sane. 

That being said I don't have a strong preference either 😆

**Comment by krzkaczor** [comment]
Released as `ts-essentials@6.0.6` 

I forgot to release it last time, sorry @quezak ! 🤦
